index.js

```diff
@@ -22,6 +24,35 @@ const STATE_MAP = {
     FAILURE: 'FAILED',
     ABORTED: 'STOPPED'
 };
+const WEBHOOK_PAGE_SIZE = 30;
+
+/**
+ * Check the status code of the server's response.
+ *
+ * If there was an error encountered with the request, this will format a human-readable
+ * error message.
+ * @method checkResponseError
+ * @param  {HTTPResponse}   response                               HTTP Response from `request` call
+ * @param  {Number}         response.statusCode                    HTTP status code of the HTTP request
+ * @param  {String}         [response.body.error.message]          Error message from the server
+ * @param  {String}         [response.body.error.detail.required]  Error resolution message
+ * @return {Promise}                                               Resolves when no error encountered.
+ *                                                                 Rejects when status code is non-200
+ */
+function checkResponseError(response) {
+    if (response.statusCode >= 200 && response.statusCode < 300) {
+        return;
+    }
+
+    const errorMessage = hoek.reach(response, 'body.error.message', {
+        default: `SCM service unavailable (${response.statusCode}).`
+    });
+    const errorReason = hoek.reach(response, 'body.error.detail.required', {
```

Comment:
@d2lam and I actually made some api calls to the hooks endpoint. We did not see the `detail` part inside the `error` object. :( Where did you find the sample response?